### PR TITLE
Treat 0- and 1-tuple symbols as tuples

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/36676

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs [hunk 1]
```diff
@@ -3228,7 +3228,7 @@ class Program
             static void validate(ModuleSymbol module)
             {
                 var globalNamespace = module.GlobalNamespace;
-                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F0").TypeWithAnnotations, new byte[] { 0 }, new byte[] { }, "System.ValueTuple");
+                VerifyBytes(globalNamespace.GetMember<FieldSymbol>("Program.F0").TypeWithAnnotations, new byte[] { 0 }, new byte[] { 0 }, "System.ValueTuple");
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs]: what is denoted by the newly introduced `0` in the metadata?

Comment (jcouv) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs]: There is an encoding position for the element name.

Comment (gafter) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs]: I don't get it.  Where could there be an element name?

---
In reply to: [359499366](https://github.com/dotnet/roslyn/pull/40459#discussion_r359499366) [](ancestors = 359499366)

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs [hunk 2]
```diff
@@ -5799,6 +5799,64 @@ public void CreateTupleTypeSymbol_BadArguments()
...
+        {
+            Item1 = item1;
+        }
+    }
+}");
+            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef, tupleComp.ToMetadataReference() });
+
+            TypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            var vt1 = comp.GetWellKnownType(WellKnownType.System_ValueTuple_T1).Construct(intType);
+            var tupleWithoutNames = comp.CreateTupleTypeSymbol(vt1, ImmutableArray.Create(new[] { (string)null }));
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs]: >tupleWithoutNames [](start = 16, length = 17)

I think this type should be identical to `vt1`.  If that's true, can you please assert that?  If not, why not?

File: src/Compilers/Core/Portable/WellKnownTypes.cs [hunk 3]
```diff
@@ -675,7 +677,7 @@ public static bool IsWellKnownType(this WellKnownType typeId)
         public static bool IsValueTupleType(this WellKnownType typeId)
```
Comment (gafter) [inline src/Compilers/Core/Portable/WellKnownTypes.cs]: >IsValueTupleType [](start = 27, length = 16)

Would you please put a comment on this that it is used for testing only?

**Comment by gafter** [comment]
Finished reviewing Iteration 1.

File: src/Compilers/Core/Portable/WellKnownTypes.cs (hunk 3 above)
Comment (jcouv) [reply src/Compilers/Core/Portable/WellKnownTypes.cs]: This method is also used in product code (`CSharpCompilation.GetWellKnownType`)

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Nullable.cs]: My bad. Fixed

**Pull request merged**